src/Core/Domain/Cart/QueryResult/CartDeliveryOption.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Cart\QueryResult;
+
+/**
+ * Holds data of cart delivery option
+ */
+class CartDeliveryOption
+{
+    /**
+     * @var int
+     */
+    private $carrierId;
+
+    /**
+     * @var string
+     */
+    private $carrierName;
+
+    /**
+     * @var string
+     */
+    private $carrierDelay;
+
+    /**
+     * @param $carrierId
+     * @param $carrierName
+     * @param $carrierDelay
+     */
+    public function __construct(int $carrierId, string $carrierName, string $carrierDelay)
+    {
+        $this->carrierId = $carrierId;
+        $this->carrierName = $carrierName;
+        $this->carrierDelay = $carrierDelay;
+    }
+
+    /**
+     * @return int
+     */
+    public function getCarrierId(): int
+    {
+        return $this->carrierId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCarrierName(): string
+    {
+        return $this->carrierName;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCarrierDelay(): string
```

Comment:
It's a delivery option, but it says getCarrier* everywhere. :confused:

Reply:
It is what delivery option consist of. Basically it is a carrier when multishipping is gone. Are you suggesting to rename it? @matks, wdyt ? It is really a question on larger scope. After multishipping feature is gone, the delivery option is basically a selection of carrier id. (then somewhere deeper in code the delivery option json is built constisting of carrier_id and delivery_address_id. Should it still be named as delivery option, or just e.g. carrierOption ?